### PR TITLE
MOJITO-906: Add delivery information

### PR DESCRIPTION
**Pull request by asdolo**
### Done
- Add wallet address section (checkbox + input) in both `BillingView` and `PaymentView`.
- Add validation for the wallet address.
- Add wallet address information / display in `ConfirmationView` with the MultiSig chip when needed.

**Note this doesn't include the integration with the backend (https://mojitoinc.atlassian.net/browse/MOJITO-972), this is just some UI elements not connected to anything.** 

### Preview

image

image

Missing screenshot for the confirmation view (need my limits rests, so I can take it now).

### CODE REVIEW

**Comment by jeisux92** [comment]
Implemented wallet delivery information in Confirmation screen.

File: app/src/lib/components/payments/DeliveryWallet/DeliveryWalletDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';
+import { CopyButton } from "../../shared/CopyButton/CopyButton";
+import { ReadOnlyWalletAddress } from "../../shared/ReadOnlyField/ReadOnlyField";
+
+export interface DeliveryWalletDetailsProps {
+    walletAddress: string,
+    isMultiSig?: boolean
+}
+const DeliveryWalletDetails: React.FC<DeliveryWalletDetailsProps> = ({
+  isMultiSig = true,
```
Comment (Danziger) [inline app/src/lib/components/payments/DeliveryWallet/DeliveryWalletDetails.tsx:11]: Hardcoded value for now as this is only UI. It's not connected to the backend yet.

File: app/src/lib/components/public/CheckoutOverlay/CheckoutOverlay.hooks.ts [hunk 2]
```diff
@@ -144,12 +151,14 @@ export function useCheckoutModalState({
   }, []);
 
   const goNext = useCallback(() => {
+    if (!isValidWalletAddress(walletAddress) && WALLET_ADDRESS_FIELD_STEPS.includes(checkoutStep)) return;
```
Comment (Danziger) [inline app/src/lib/components/public/CheckoutOverlay/CheckoutOverlay.hooks.ts:154]: I'm not really sure I like the idea of doing the validation at this level, but it's ok for now.

File: app/src/lib/components/public/SuccessOverlay/SuccessOverlay.tsx [hunk 3]
```diff
@@ -30,8 +30,8 @@ export const PUISuccessOverlay: React.FC<PUISuccessOverlayProps> = ({
   const isPathname = isUrlPathname(url);
 
   useLayoutEffect(() => {
-    if (purchaseSuccess && isPathname) {
-      persistReceivedRedirectUri3DS(window.location.href);
+    if (purchaseSuccess) {
+      if (isPathname) persistReceivedRedirectUri3DS(window.location.href);
```
Comment (Danziger) [inline app/src/lib/components/public/SuccessOverlay/SuccessOverlay.tsx:34]: Stay on the page (waiting for the automatic redireciton) but do not persist anything if `url ` is a not a pathname.

File: app/src/lib/components/shared/ReadOnlyField/ReadOnlyField.tsx [hunk 4]
```diff
@@ -53,6 +52,34 @@ export const ReadOnlyCardField: React.FC<TextFieldProps> = styled((props) => (
   },
   "& .MuiInputBase-input": {
     "&.Mui-disabled": {
+      ...typography.body1,
```
Comment (Danziger) [inline app/src/lib/components/shared/ReadOnlyField/ReadOnlyField.tsx:55]: Having these components and styles like this is bad, we should move everything to the theme (using custom variants) as part of https://mojitoinc.atlassian.net/browse/MOJITO-794.

**Review by Danziger — APPROVED** [review]

**Pull request merged**